src/cpp/tests/testthat/test-dataviewer.R

```diff
@@ -0,0 +1,33 @@
+#
+# test-dataviewer.R
+#
+# Copyright (C) 2022 by RStudio, PBC
+#
+# Unless you have received this program directly from RStudio pursuant
+# to the terms of a commercial license agreement with RStudio, then
+# this program is licensed to you under the terms of version 3 of the
+# GNU Affero General Public License. This program is distributed WITHOUT
+# ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+# MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+# AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+#
+#
+
+context("data viewer")
+
+test_that(".rs.formatDataColumnDispatch() iterates over the classes", {
+    registerS3method("[", "foo",  function(x, ...) {
```

Comment:
cc @DavisVaughan you might have views on this.

Reply:
I assume that `getS3Method()` looks in the S3MethodsTable for the method, and I don't _think_ that just sourcing a method will put it there, but I do think `registerS3Method()` will add it to the table, so that may explain it

Reply:
Yeah, and I suppose what we're after here are methods defined in packages, which would have been correctly registered. So I think the pr is in good shape.